packages/sources/nomics-test/src/crypto-utils.ts

```diff
@@ -0,0 +1,153 @@
+import { customSettings } from './config'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+
+export const inputParameters = {
+  base: {
+    aliases: ['from', 'coin'],
+    required: true,
+    type: 'string',
+    description: 'The symbol of symbols of the currency to query',
+  },
+  quote: {
+    aliases: ['to', 'market', 'convert'],
+    required: true,
+    type: 'string',
+    description: 'The symbol of the currency to convert to',
+  },
+} as const
+
+export interface RequestParams {
+  base: string
+  quote: string
+}
+
+export interface ResponseSchema {
+  id: string
+  currency: string
+  symbol: string
+  name: string
+  logo_url: string
+  status: string
+  price: string
+  price_date: string
+  price_timestamp: string
+  circulating_supply: string
+  max_supply: string
+  market_cap: string
+  num_exchanges: string
+  num_pairs: string
+  num_pairs_unmapped: string
+  first_candle: string
+  first_trade: string
+  first_order_book: string
+  rank: string
+  rank_delta: string
+  high: string
+  high_timestamp: string
+  '1d': {
+    volume: string
+    price_change: string
+    price_change_pct: string
+    volume_change: string
+    volume_change_pct: string
+    market_cap_change: string
+    market_cap_change_pct: string
+  }
```

Comment:
Can we extract this into its own type and re-use that? Assuming it's the same payload for each of these time ranges